source/tyr/tyr/resources.py

```diff
@@ -550,3 +569,119 @@ def delete(self, id):
+        parser.add_argument('max_bss_duration_to_pt', type=parser_args_type.float_gt_0, required=False,
+                            case_sensitive=False)
+        parser.add_argument('max_car_duration_to_pt', type=parser_args_type.float_gt_0, required=False,
+                            case_sensitive=False)
+        parser.add_argument('first_section_mode[]',
+                            type=parser_args_type.option_value(['walking', 'car', 'bss', 'bike']),
+                            required=False, action='append', case_sensitive=False, dest='first_section_mode')
+        parser.add_argument('last_section_mode[]',
+                            type=parser_args_type.option_value(['walking', 'car', 'bss', 'bike']),
+                            required=False, action='append', case_sensitive=False, dest='last_section_mode')
+        self.args = parser.parse_args()
+
+    def check_resources(f):
+        @wraps(f)
+        def wrapper(*args, **kwds):
+            tp = kwds.get('traveler_type')
+            if tp in acceptable_traveler_types:
+                return f(*args, **kwds)
+            return {'error': 'traveler profile: {0} is not one of in {1}'.format(tp, acceptable_traveler_types)}, 400
+        return wrapper
+
+    @marshal_with(traveler_profile)
+    @check_resources
+    def get(self, name=None, traveler_type=None):
+        try:
+            traveler_profiles = []
+            # If traveler_type is not specified, we return all existent traveler profiles of this instance
+            if traveler_type is None:
+                traveler_profiles += models.TravelerProfile.get_all_by_coverage(coverage=name)
+            else:
+                profile = models.TravelerProfile.get_by_coverage_and_type(coverage=name, traveler_type=traveler_type)
+                if profile:
+                    traveler_profiles.append(profile)
+
+            if traveler_profiles:
+                return traveler_profiles
+            return {'error': 'No matching traveler profiles are found in db'}, 404
+        except (sqlalchemy.exc.IntegrityError, sqlalchemy.orm.exc.FlushError), e:
+            return {'error': str(e)}, 409
+        except Exception:
+            logging.exception("fail")
+            raise
+
+    @marshal_with(traveler_profile)
+    @check_resources
+    def post(self, name=None, traveler_type=None):
+        try:
+            instance = models.Instance.get_by_name(name)
+            if instance is None:
+                return {'error': "Coverage: {0} doesn't exist".format(name)}
+            profile = models.TravelerProfile()
+            profile.coverage_id = instance.id
+            for (attr, default_value) in default_traveler_profile_params[traveler_type].iteritems():
+                # override hardcoded values by args if args are not None
+                value = default_value if self.args.get(attr) is None else self.args.get(attr)
+                setattr(profile, attr, value)
+            db.session.add(profile)
```

Comment:
the `traveler_type` is in `default_traveler_profile_params`, even though `traveler_type` is no longer in args, it'll still be set. :)